11_FFTBloom/main.cpp

```diff
@@ -965,58 +979,140 @@ class FFTBloomApp final : public application_templates::MonoDeviceApplication, p
-		// We need to pass the log of number of workgroups as a push constant now
-		uint32_t numWorkgroups = core::roundUpToPoT(marginSrcDim.height) / 2;
-		uint32_t workgroupsLog2 = std::bit_width(numWorkgroups) - 1;
-		m_computeCmdBuf->pushConstants(m_lastAxisFFT_convolution_lastAxisIFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, numWorkgroupsLog2), sizeof(pushConstants.numWorkgroupsLog2), &workgroupsLog2);
-
-		m_computeCmdBuf->dispatch(numWorkgroups, 1, 1);
+		cmdBuf->bindComputePipeline(m_lastAxisFFT_convolution_lastAxisIFFTPipeline.get());
+		// Update padding for run along rows
+		cmdBuf->pushConstants(m_firstAxisFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, padding), sizeof(paddingAlongRows), &paddingAlongRows);
+		// One workgroup per row in the lower half of the DFT
+		cmdBuf->dispatch(core::roundUpToPoT(m_marginSrcDim.height) / 2, 1, 1);
 
 		// Recycle pipeline barrier, only have to change which buffer we need to wait to be written to
 		bufBarrier.range.buffer = m_rowMajorBuffer;
-		m_computeCmdBuf->pipelineBarrier(asset::E_DEPENDENCY_FLAGS(0), bufferPipelineBarrierInfo);
+		cmdBuf->pipelineBarrier(asset::E_DEPENDENCY_FLAGS(0), bufferPipelineBarrierInfo);
 
 		// Finally run the IFFT on the first axis
-		m_computeCmdBuf->bindComputePipeline(m_firstAxisIFFTPipeline.get());
-		m_computeCmdBuf->bindDescriptorSets(asset::EPBP_COMPUTE, m_firstAxisIFFTPipeline->getLayout(), 0, 1, &m_firstAxisIFFTDescriptorSet.get());
+		cmdBuf->bindComputePipeline(m_firstAxisIFFTPipeline.get());
+		// Update padding for run along columns
+		cmdBuf->pushConstants(m_firstAxisFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, padding), sizeof(paddingAlongColumns), &paddingAlongColumns);
 		// One workgroup per 2 columns
-		m_computeCmdBuf->dispatch(srcDim.width / 2, 1, 1);
-		m_computeCmdBuf->end();
+		cmdBuf->dispatch(srcDim.width / 2, 1, 1);
+
+		// -------------------------------------- DRAW END ----------------------------------------
+
+		// BLIT
+		{
+			auto swapImg = m_surface->getSwapchainResources()->getImage(m_currentImageAcquire.imageIndex);
+
+			using image_memory_barrier_t = IGPUCommandBuffer::SImageMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier>;
+			image_memory_barrier_t imgComputeToBlitBarrier = {
+				.barrier = {
+					.dep = {
+						.srcStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT,
+						.srcAccessMask = ACCESS_FLAGS::STORAGE_WRITE_BIT | ACCESS_FLAGS::STORAGE_READ_BIT,
+						.dstStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+						.dstAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT
+					}
+			},
+			.image = m_outImgView->getCreationParameters().image.get(),
+			.subresourceRange = whole2DColorImage,
+			.oldLayout = IImage::LAYOUT::GENERAL,
+			.newLayout = IImage::LAYOUT::GENERAL
```

Comment:
if you want no layout transition, you can keep them uninitialized, both being same layout prevents transition (UNDEFINED and UNDEFINED)